.travis.yml

```diff
@@ -16,26 +23,93 @@ matrix:
       env: LUA="luajit 2.0"
     - compiler: ": LuaJIT21"
       env: LUA="luajit 2.1"
+    - env: OS=el DIST=6 PRODUCT=tarantool-luacheck
+    - env: OS=el DIST=7 PRODUCT=tarantool-luacheck
+    - env: OS=fedora DIST=26 PRODUCT=tarantool-luacheck
+    - env: OS=fedora DIST=27 PRODUCT=tarantool-luacheck
 
 install:
-  - pip install --user hererocks
-  - pip install --user codecov
-  - hererocks here --$LUA -r https://github.com/mpeterv/luarocks@upgrade-install
-  - export PATH="$PWD/here/bin:$PATH"
-  - luarocks install lanes --upgrade --upgrade-deps
-  - luarocks install busted --upgrade --upgrade-deps
-  - luarocks install cluacov --upgrade --upgrade-deps
-
+  - |
+    if [ -n "${LUA}" ]; then
+        pip install --user hererocks
+        pip install --user codecov
+        sudo apt-get install luarocks
+        hererocks here --$LUA -r https://github.com/mpeterv/luarocks@upgrade-install
+        export PATH="$PWD/here/bin:$PATH"
+        luarocks install lanes --upgrade --upgrade-deps
+        luarocks install busted --upgrade --upgrade-deps
+        luarocks install cluacov --upgrade --upgrade-deps
+    fi;
 script:
-  - busted -c
-  - lua -e 'package.path="./src/?.lua;./src/?/init.lua;"..package.path' -lluacov bin/luacheck.lua luacheck-dev-1.rockspec -j2
-  - lua -e 'package.preload.lfs=error;package.path="./src/?.lua;./src/?/init.lua;"..package.path' -lluacov bin/luacheck.lua src | grep 'I/O error'
-  - lua -e 'package.preload.lanes=error;package.path="./src/?.lua;./src/?/init.lua;"..package.path' -lluacov bin/luacheck.lua --version | grep 'Not found'
-  - lua install.lua path/to/luacheck
-  - mv src src2
-  - path/to/luacheck/bin/luacheck spec/*.lua
-  - mv src2 src
+  - |
+    if [ -n "${LUA}" ]; then
+        busted -c
+        lua -e 'package.path="./src/?.lua;./src/?/init.lua;"..package.path' \
+            -lluacov bin/luacheck.lua luacheck-dev-1.rockspec -j2
+        lua -e 'package.preload.lfs=error;package.path="./src/?.lua;./src/?/init.lua;"..package.path' \
+            -lluacov bin/luacheck.lua src | grep 'I/O error'
+        lua -e 'package.preload.lanes=error;package.path="./src/?.lua;./src/?/init.lua;"..package.path' \
+            -lluacov bin/luacheck.lua --version | grep 'Not found'
+        lua install.lua path/to/luacheck
+        mv src src2
+        path/to/luacheck/bin/luacheck spec/*.lua
+        mv src2 src
+    else
+        # Luacheck has a major.minor.patch versioning with unannotated tags.
+        # Packpack can't handle it automaticaly, because it expects
+        # major.minor annotated tags. We had to pass version by setting VERSION.
```

Comment:
packpack correctly handle major.minor.patch and likely more complex tag schemas. I think the question here is just that tags are not annotated.

Sorry for nitpicking, but I’m againts of confusing comments.

Reply:
You are right. Fixed here and also in lulpeg and lrexlib.